driver-mapping/src/main/java/com/datastax/driver/mapping/Mapper.java

```diff
@@ -78,17 +81,18 @@ Session session() {
         return manager.getSession();
     }
 
-    PreparedStatement getPreparedQuery(QueryType type) {
-        PreparedStatement stmt = preparedQueries.get(type);
+    PreparedStatement getPreparedQuery(QueryType type, Option... options) {
+        String queryString;
+        queryString = type.makePreparedQueryString(tableMetadata, mapper, options);
```

Comment:
This rebuilds the whole query for each save/delete call :(
I think it would be better to use a special key class.